odoo/sql_db.py

```diff
@@ -245,9 +255,23 @@ def execute(self, query, params=None, log_exceptions=None):
             encoding = psycopg2.extensions.encodings[self.connection.encoding]
             _logger.debug("query: %s", self._obj.mogrify(query, params).decode(encoding, 'replace'))
         now = time.time()
+        catch_traceback_query = os.environ.get("catch_traceback_query", "")
         try:
             params = params or None
             res = self._obj.execute(query, params)
+            if catch_traceback_query:
+                print(self._obj.query)
+                for query_to_catch in QUERY_TO_CATCHS:
+                    query = self._obj.query.replace(b"\n", b" ").replace(b"\'", b'"').replace(b" ", b"")
+                    if query_to_catch.lower().replace(b" ", b"") in query.lower():
+                        tcbk = traceback.extract_stack()
+                        key = hashlib.md5(repr(tcbk).encode('UTF-8') + query_to_catch).hexdigest()
+                        TRACEBACK_GROUP[key]['count'] += 1
+                        TRACEBACK_GROUP[key]['tcbk'] = tcbk
+                        tcbk_str = ""
+                        for key, tcbk in sorted(TRACEBACK_GROUP.items(), key=lambda x: x[1]['count'], reverse=True):
+                            tcbk_str += "%d,%s,%s\n" % (tcbk['count'], '\n,'.join(map(lambda a: repr(a), tcbk['tcbk'])), query_to_catch)
+                        _logger.info("query executed: %s", self._obj.query, exc_info=True)
```

Comment:
Could you check if the logger shows the same than 

tcbk variable content

And the tcbk_str is not needed anymore since that it was used to generate the full tmp txt but this time we are using only logger so the old cases don't need to be cumulated

Reply:
@moylop260 the _logger only print the string "query executed: %s", not include the traceback (was set as Nonetype: None),
```zsh
2022-01-13 23:01:41,985 44342 INFO qfg130 odoo.sql_db: query executed: b'SELECT "sale_order".id FROM "sale_order" WHERE ("sale_order"."state" in (\'sale\',\'done\')) ORDER BY "sale_order"."id"  ' 
NoneType: None
```

 after check on stackoverflow https://stackoverflow.com/questions/1508467/log-exception-with-traceback-in-python

If I include

```
try:
    raise Exception(tcbk_str)
except:
    _logger.info("query executed: %s", raw_query, exc_info=True)
```

when the query it's catched this going printed:

```zsh
2022-01-13 22:58:01,842 43685 INFO qfg130 odoo.sql_db: query executed: b'SELECT "sale_order".id FROM "sale_order" WHERE ("sale_order"."state" in (\'sale\',\'done\')) ORDER BY "sale_order"."id"  ' 
Traceback (most recent call last):
  File "/home/frahik/ODOO/MODULOS/odoo/odoo/sql_db.py", line 281, in execute
    raise Exception(tcbk_str)
Exception: 1,<FrameSummary file ./odoo-bin, line 8 in <module>>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/cli/command.py, line 60 in main>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/cli/shell.py, line 125 in run>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/cli/shell.py, line 118 in shell>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/cli/shell.py, line 82 in console>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/cli/shell.py, line 91 in ipython>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/__init__.py, line 126 in start_ipython>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/traitlets/config/application.py, line 846 in launch_instance>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/terminal/ipapp.py, line 356 in start>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py, line 563 in mainloop>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py, line 554 in interact>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py, line 2901 in run_cell>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py, line 2947 in _run_cell>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/async_helpers.py, line 68 in _pseudo_sync_runner>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py, line 3172 in run_cell_async>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py, line 3364 in run_ast_nodes>
,<FrameSummary file /home/frahik/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py, line 3444 in run_code>
,<FrameSummary file <ipython-input-1-bfa0db232c63>, line 1 in <module>>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/models.py, line 1654 in search>
,<FrameSummary file /home/frahik/ODOO/CLIENTES/qualifirst/customer_portal_hide_negative/models/sale_order.py, line 13 in _search>
,<FrameSummary file /home/frahik/ODOO/CLIENTES/qualifirst/customer_portal_force_date_from/models/sale_order.py, line 12 in _search>
,<FrameSummary file /home/frahik/ODOO/CLIENTES/qualifirst/qfg/models/models.py, line 34 in _search>
,<FrameSummary file /home/frahik/ODOO/CLIENTES/qualifirst/search_action_domain/models/models.py, line 37 in _search>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/models.py, line 4512 in _search>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/sql_db.py, line 185 in wrapper>
,<FrameSummary file /home/frahik/ODOO/MODULOS/odoo/odoo/sql_db.py, line 272 in execute>,b'SELECT "sale_order".id FROM "sale_order" WHERE ("sale_order"."state" in ("sale", "done")) ORDER BY "sale_order"."id"'
```

but we need keep tcbk_str to be printed :grinning: